src/main/java/org/jenkinsci/plugins/workflow/libs/SCMSourceRetriever.java

```diff
@@ -75,13 +75,13 @@ public SCMSource getScm() {
         if (revision == null) {
             throw new AbortException("No version " + version + " found for library " + name);
         }
-        doRetrieve(scm.build(revision.getHead(), revision), revision.isDeterministic(), target, run, listener);
+        doRetrieve(scm.build(revision.getHead(), revision), target, run, listener);
     }
 
-    static void doRetrieve(@Nonnull SCM scm, boolean deterministic, FilePath target, Run<?, ?> run, TaskListener listener) throws Exception {
+    static void doRetrieve(@Nonnull SCM scm, FilePath target, Run<?, ?> run, TaskListener listener) throws Exception {
         // Adapted from CpsScmFlowDefinition:
         SCMStep delegate = new GenericSCMStep(scm);
-        delegate.setPoll(!deterministic); // TODO is this desirable?
+        delegate.setPoll(false); // TODO we have no API for determining if a given SCMHead is branch-like or tag-like; would we want to turn on polling if the former?
```

Comment:
Yeah you could argue it either way. With a `checkout` step the user chooses, but the default is to poll. The argument to poll (when the SCM refers to something like a branch) is that if the library does change, and you were to do another build, it might behave differently, which is something you probably want to know earlier rather than later. On the other hand, you might be making a very minor change to a library with hundreds of usages.